src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dns/ClusterDnsCache.scala

```diff
@@ -51,41 +40,29 @@ object ClusterDnsCache {
   * @param proxyConfig the proxy configuration
   * @param dbRef provides access to the database
   */
-class ClusterDnsCache(proxyConfig: ProxyConfig, dbRef: DbReference) extends Actor with LazyLogging {
-  var ProjectNameToHost: Map[(GoogleProject, ClusterName), GetClusterResponse] = Map.empty
-
-  import context.dispatcher
-
-  override def preStart(): Unit = {
-    super.preStart()
-    self ! RefreshFromDatabase
-  }
-
-  override def receive: Receive = {
-    case RefreshFromDatabase =>
-      queryForClusters pipeTo self
-
-    case ProcessClusters(clusters) =>
-      processClusters(clusters)
-      scheduleRefresh
-
-    // add one ready cluster to the caches now instead of waiting for the refresh cycle
-    case ProcessReadyCluster(cluster) =>
-      sender ! processReadyCluster(cluster)
-
-    case GetByProjectAndName(googleProject, clusterName) =>
-      sender ! ProjectNameToHost.getOrElse((googleProject, clusterName), ClusterNotFound)
-  }
-
-  private def scheduleRefresh = {
-    context.system.scheduler.scheduleOnce(proxyConfig.dnsPollPeriod, self, RefreshFromDatabase)
-  }
-
-  private def queryForClusters: Future[ProcessClusters] ={
-    dbRef.inTransaction { dataAccess =>
-      dataAccess.clusterQuery.listActive()
-    }.map(ProcessClusters.apply)
-  }
+class ClusterDnsCache(proxyConfig: ProxyConfig, dbRef: DbReference)(implicit  executionContext: ExecutionContext) extends LazyLogging {
+
+  val ProjectNameToHost = CacheBuilder.newBuilder()
+    .expireAfterWrite(5, TimeUnit.SECONDS)
+    .maximumSize(100)
+    .build(
+      new CacheLoader[(GoogleProject, ClusterName), Future[GetClusterResponse]] {
+        def load(key: (GoogleProject, ClusterName)) = {
+          dbRef.inTransaction {dataAccess => dataAccess.clusterQuery.getActiveClusterByName(key._1, key._2)}.map{
+            c => c match {
+              case  Some(cluster) => {
+                val res = projectNameToHostEntry(cluster)._2
```

Comment:
If this is still relevant after refactoring based on the other comments, we may want to rewrite as something like

```scala
projectNameToHostEntry(cluster) match { 
  case (_, ClusterReady(host)) => ??? 
}
```

Reply:
I'd still recommend using pattern matching like ^^.